stencil-library/src/components/details/details.tsx

```diff
@@ -0,0 +1,97 @@
+import { Component, Host, h, Prop } from '@stencil/core';
+import { ErrorState } from './utils';
+
+interface IEntityProps {
+  title: string;
+  entityHeadInfo: {
+    title: string,
+    value: any
+  }[];
+  entitySections: {
```

Comment:
Instead of passing all of this config I think we should leverage named slots and subcomponents such as `DetailSectionTitle` etc

Reply:
Will take a look and see if I find a better approach.